source/Hardpoint.h

```diff
@@ -101,6 +103,7 @@ class Hardpoint {
 private:
 	// The weapon installed in this hardpoint.
 	const Outfit *outfit = nullptr;
+	const Weapon *weapon = nullptr;
```

Comment:
Why store the weapon? Any reason why we can't use `outfit->GetWeapon()` where we need to access it?

I didn't check, but maybe `GetOutfit()` is unused and not needed in which case you remove `outfit` instead.